src/libs/websocket.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+
+import { ApiGatewayManagementApiClient, PostToConnectionCommand } from '@aws-sdk/client-apigatewaymanagementapi';
+import util from 'util';
+
+import { WsConnectionInfo } from '@models/websocket';
+import { ServerlessMysql } from 'serverless-mysql';
+
+export const DEFAULT_API_GATEWAY_RESPONSE: APIGatewayProxyResult = {
+  statusCode: 200,
+  body: '',
+};
+
+/**
+ * Extracts the connection identifier from the API Gateway proxy event,
+ * and adds the url according to the current WS_DOMAIN configuration
+ */
+export const connectionInfoFromEvent = (
+  event: APIGatewayProxyEvent,
+): WsConnectionInfo => {
+  const connID = event.requestContext.connectionId;
+
+  // This will enter when running the service on serverless offline mode
+  if (process.env.IS_OFFLINE === 'true') {
+    return {
+      id: connID,
+      url: 'http://localhost:3002', // TODO: Fetch the port number from the config
+    };
+  }
+
+  // Obtaining the domain URL from the server configurations
+  const domain = process.env.WS_DOMAIN;
+  if (!domain) {
+    // Throw so we receive an alert telling us that something is wrong with the env variable
+    // instead of trying to invoke a lambda at https://undefined
+    throw new Error('[ALERT] Domain not on env variables');
+  }
+
+  return {
+    id: connID,
+    url: util.format('https://%s', domain),
+  };
+};
+
+/**
+ * Initializes a websocket connection by storing its identifier on the database
+ */
+export const initWsConnection = async (
+  client: ServerlessMysql,
+  connInfo: WsConnectionInfo,
+): Promise<string> => {
+  /* There is a small chance that the connectionId will be duplicated between $connect
+   * requests. When this happens we can safely ignore the insert.
+   */
+  await client.query(
+    `INSERT INTO websockets (connection, url)
+              VALUES (?, ?)
+              ON DUPLICATE KEY UPDATE connection = connection;`,
+    [connInfo.id, connInfo.url],
+  );
+
+  return connInfo.id;
+};
+
+/**
+ * Ends a websocket connection by removing its identifier from the database
+ */
+export const endWsConnection = async (
```

Comment:
Should we have a "cleanup" routine to make sure connections are being removed from database in case we miss the connection end message?

Reply:
Yes, we will in a future PR.
Our expectation is to have a daily cleanup of all connections that have been last updated more than 10 minutes ago.